s6/debian-root/etc/cont-init.d/20-start.sh

```diff
@@ -14,7 +14,22 @@ $bashCmd /start.sh
 if [ -n "$PYTEST" ]; then
     sed -i 's/^gravity_spinup$/#gravity_spinup # DISABLED FOR PYTEST/g' "$(which gravity.sh)"
 fi
-if [ -z "$SKIPGRAVITYONBOOT" ] || [ ! -f /etc/pihole/gravity.db ]; then
+
+# gravity.sh consistent check for gravityDBfile's existence
+# GRAVITYDB may be overwritten by source pihole-FTL.conf below
+GRAVITYDB="/etc/pihole/gravity.db"
+
+# Source pihole-FTL from install script
+pihole_FTL="/etc/pihole/pihole-FTL.conf"
+if [[ -f "${pihole_FTL}" ]]; then
+  source "${pihole_FTL}"
```

Comment:
Sourcing `pihole-FTL.conf` is bad because of mainly two things:

1. FTL allows for a multitude of comment styles, not only `#` at the beginning of a line.
  There are many users using `;` for comments because this is in line with that PHP's INI parsing expects. However, sourcing a file with such comments in it (with `root` privileges) can be very dangerous.
2. Pi-hole v6.0 will switch to [TOML](https://toml.io/) for the config file and `source` will not work as expected.

Reply:
Regarding point 1.What would be the recommended way of checking the `GravityDB` setting in pihole-FTL.conf?
I'm following how it's normally checked in gravity.sh: [gravity.sh#L69](https://github.com/pi-hole/pi-hole/blob/4736e03108763cc2d5659f48d8a1e8a64d9b2608/gravity.sh#L69)

Does pihole -g/pihole updateGravity internally run as a different user? Near as I can tell, docker-pi-hole currently touches that code path in gravity as root user currently: [20-start.sh#L18](https://github.com/pi-hole/docker-pi-hole/blob/76be304d79d8c8820575cf68128e0e6b53b57d82/s6/debian-root/etc/cont-init.d/20-start.sh#L18)

Regarding point 2. In the original issue, I floated the idea of splitting out the initialization responsibilities from gravity.sh into it's own script mainly for this reason, to avoid maintaining the same functionality in both repos. 

In my head this would look like: refactoring gravity.db and local.list initialization into an `init_gravity.sh` located under `pi-hole/advanced/Scripts` and touching [`PI_HOLE_FILES` in `basic-install.sh`](https://github.com/pi-hole/pi-hole/blob/4736e03108763cc2d5659f48d8a1e8a64d9b2608/automated%20install/basic-install.sh#L73) to make sure [it's properly cleaned up](https://github.com/pi-hole/pi-hole/blob/4736e03108763cc2d5659f48d8a1e8a64d9b2608/automated%20install/basic-install.sh#L1427). Both gravity and docker could source the file that gets copied to the installation directory.

Reply:
You could use something like
``` bash
GRAVITYDB="$(grep --color=never -Po "^GRAVITYDB=\K.*" "/etc/pihole/pihole-FTL.conf" 2> /dev/null || echo "/etc/pihole/gravity.db")"
```

which would also make the `if` unnecessary as the `|| echo` will account for a non-existing conf file. True, this would not (yet) account for a future TOML transition, but that's a problem of when this will happen. It's probably better to add something like
```
pihole-FTL which gravitydb
```
to get the answer independently from the particular language of the config file.

As for the other parts of the code: It's probably just done wrong there.

Reply:
```
pihole-FTL which gravitydb
```

Errors out in my testing, running it in the container yields this output:
> root@78540f0a8f52:/# pihole-FTL which gravitydb
pihole-FTL: invalid option -- 'which'
Command: 'pihole-FTL which gravitydb'
Try 'pihole-FTL --help' for more information

Reading the last match off `/var/log/pihole-FTL.log` might be a way around the config file issue//eventual TOML transition, assuming it's stable enough to rely on:

```
gravityDBfile="$(grep --color=never -Po "GRAVITYDB: Using \K.*" "/var/log/pihole-FTL.log" 2> /dev/null | tail -1 || echo "/etc/pihole/gravity.db")"
```

Reply:
> It's probably better **to add** something like
>
> ```
> pihole-FTL which gravitydb
> ```

Sorry, I see this was misleading - with *add* I meant we can add this to `pihole-FTL`. There is no such feature at the moment.

---

> Reading the last match off `/var/log/pihole-FTL.log` might be a way around the config file issue

I'm not familiar enough (not at all TBH) with this script is used. Is it guaranteed that FTL was started before (and had time to parse its config file)? Furthermore, is this script run only once (immediately at start) or can it be run also later on? In the latter case, `pihole-FTL.log` might have been rotated away and the information you are looking for is already gone.

Reply:
Kinda like:

![image](https://user-images.githubusercontent.com/1998970/126881371-95cbb70e-2304-45e8-be34-e927ce77f47e.png)

-----

The script runs on container start,  but I wouldn't rely on grepping the log file for the location of `gravity.db`, I would just grep the config file...

Reply:
😅 I just forced pushed the log based solution before reading this. Since FTL is started before and brought down after this check (originally because it was needed to run gravity before gravity was relegated to a cron job), it shouldn't be rotated away and should satisfy @DL6ER's concerns. 

I guess what I can do is output a warning if pihole-FTL.conf is not found, so that this isn't forgotten when it's replaced with TOML and grep the log file. I'll test and force push that shortly

Reply:
> I just forced pushed the log based solution before reading this

But this is not what is in your code. Is this correct?

Reply:
Correct, I pushed the log based solution just prior to reading PromoFaux's feedback:

> I wouldn't rely on grepping the log file for the location of gravity.db, I would just grep the config file...

And  updated again to reflect his preference to grep the config file; I initially didn't want to do that given it'll silently regress if/when it's moved to TOML, but added a line to the output to mention which config file is being grepped so the regression will be less silent if we do forget to update this in lockstep.